imports/plugins/included/taxes-avalara/client/containers/avalaraSettingsFormContainer.js

```diff
@@ -0,0 +1,153 @@
+        }
+        return Alerts.toast(i18next.t("settings.testCredentialsFailed"), "error");
+      } catch (e) {
+        return Alerts.toast(i18next.t("settings.testCredentialsFailed"), "error");
+      }
+    });
+  }
+};
+
+const countryDefaults = ["US", "CA"];
+let validCountries = null;
+
+const composer = (props, onData) => {
+  const shownFields = {
+    ["settings.avalara.apiLoginId"]: AvalaraPackageConfig._schema["settings.avalara.apiLoginId"],
+    ["settings.avalara.username"]: AvalaraPackageConfig._schema["settings.avalara.username"],
+    ["settings.avalara.password"]: AvalaraPackageConfig._schema["settings.avalara.password"],
+    ["settings.avalara.companyCode"]: AvalaraPackageConfig._schema["settings.avalara.companyCode"],
+    ["settings.avalara.shippingTaxCode"]: AvalaraPackageConfig._schema["settings.avalara.shippingTaxCode"],
+    ["settings.addressValidation.enabled"]: AvalaraPackageConfig._schema["settings.addressValidation.enabled"],
+    ["settings.addressValidation.countryList"]: AvalaraPackageConfig._schema["settings.addressValidation.countryList"],
+    ["settings.avalara.requestTimeout"]: AvalaraPackageConfig._schema["settings.avalara.requestTimeout"],
+    ["settings.avalara.mode"]: AvalaraPackageConfig._schema["settings.avalara.mode"],
+    ["settings.avalara.performTaxCalculation"]: AvalaraPackageConfig._schema["settings.avalara.performTaxCalculation"],
+    ["settings.avalara.enableLogging"]: AvalaraPackageConfig._schema["settings.avalara.enableLogging"],
+    ["settings.avalara.logRetentionDuration"]: AvalaraPackageConfig._schema["settings.avalara.logRetentionDuration"],
+    ["settings.avalara.commitDocuments"]: AvalaraPackageConfig._schema["settings.avalara.commitDocuments"]
+  };
+  const hiddenFields = [
+    "settings.avalara.enabled",
+    "settings.avalara.companyId"
+  ];
+
+  validCountries = validCountries || Countries.find({ value: { $in: countryDefaults } }).fetch();
+
+  const fieldsProp = {
+    "settings.addressValidation.countryList": {
+      renderComponent: "multiselect",
+      options: validCountries,
+      defaultValue: _.map(validCountries, "value")
+    },
+    "settings.avalara.mode": {
+      renderComponent: "select",
+      options: [
+        { value: true, label: "Production Mode" },
+        { value: false, label: "Testing-Sandbox Mode" }],
+      label: undefined
+    },
+    "settings.avalara.logRetentionDuration": {
+      renderComponent: "string",
+      inputType: "number"
+    },
+    "settings.avalara.requestTimeout": {
+      renderComponent: "string",
+      inputType: "number"
+    },
+    "settings.avalara.password": {
+      inputType: "password"
+    }
+  };
+
+  const logFieldsProp = {
+    data: {
+      renderComponent: "string",
+      multiline: true,
+      maxRows: 15
+    },
+    date: {
+      disabled: true
+    }
+  };
+
+  const shownLogFields = {
+    date: LogSchema._schema.date,
+    data: LogSchema._schema.data
+  };
```

Comment:
Everything above here in this function is static. Can you remove it from here and use `withProps` to pass them in instead? Then this composer function would only set `settings` prop whenever it reruns.